# Análisis exploratorio inicial — Telco Customer Churn

Este notebook realiza la exploración inicial del dataset utilizado en el proyecto Sistema-MLOps.

El objetivo es identificar la estructura general de los datos, cantidad de registros y columnas, tipos de datos, valores nulos y distribución de la variable objetivo `Churn`.

In [9]:
import pandas as pd

df = pd.read_csv("../data/raw/churn.csv")

print("Dataset cargado correctamente")
print("Filas y columnas:", df.shape)

Dataset cargado correctamente
Filas y columnas: (7043, 21)


In [10]:
print("Número de filas:", df.shape[0])
print("Número de columnas:", df.shape[1])

Número de filas: 7043
Número de columnas: 21


In [11]:
df.head()

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


In [12]:
print("Tipos de datos:")
print(df.dtypes)

Tipos de datos:
customerID              str
gender                  str
SeniorCitizen         int64
Partner                 str
Dependents              str
tenure                int64
PhoneService            str
MultipleLines           str
InternetService         str
OnlineSecurity          str
OnlineBackup            str
DeviceProtection        str
TechSupport             str
StreamingTV             str
StreamingMovies         str
Contract                str
PaperlessBilling        str
PaymentMethod           str
MonthlyCharges      float64
TotalCharges            str
Churn                   str
dtype: object


In [13]:
print("Valores nulos por columna:")
print(df.isnull().sum())

Valores nulos por columna:
customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64


In [14]:
print("Distribución de Churn:")
print(df["Churn"].value_counts())

Distribución de Churn:
Churn
No     5174
Yes    1869
Name: count, dtype: int64


In [15]:
print(df.isnull().sum())

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64


In [16]:
print("Distribución porcentual de Churn:")
print((df["Churn"].value_counts(normalize=True) * 100).round(2))

Distribución porcentual de Churn:
Churn
No     73.46
Yes    26.54
Name: proportion, dtype: float64


## Conclusiones de la exploración inicial

- El dataset contiene 7043 registros y 21 columnas.
- La variable objetivo es `Churn`.
- La variable objetivo contiene las categorías `No` y `Yes`.
- La revisión inicial con Pandas no identificó valores nulos en las columnas.
- La distribución de la variable objetivo corresponde a 5174 registros con `Churn = No` y 1869 registros con `Churn = Yes`.
- Se identificaron los tipos de datos y la estructura general del dataset.
- Los resultados de esta exploración servirán como base para las siguientes etapas de limpieza y preparación de los datos.